app/src/main/java/org/fossasia/openevent/app/core/track/list/viewholder/TracksViewHolder.java

```diff
@@ -0,0 +1,31 @@
+package org.fossasia.openevent.app.core.track.list.viewholder;
+
+import android.support.v7.widget.RecyclerView;
+
+import org.fossasia.openevent.app.data.tracks.Track;
+import org.fossasia.openevent.app.databinding.TracksItemBinding;
+
+public class TracksViewHolder extends RecyclerView.ViewHolder {
+    private final TracksItemBinding binding;
+
+    public TracksViewHolder(TracksItemBinding binding) {
+        super(binding.getRoot());
+        this.binding = binding;
+    }
+
+    public void bind(Track track) {
+        if (!validColor(track.color)) {
+            track.color = null;
+        }
+        if (!validColor(track.fontColor)) {
+            track.fontColor = null;
+        }
+        binding.setTrack(track);
+        binding.executePendingBindings();
+    }
+
+    private boolean validColor(String color) {
+        String pattern = "^#[0-9A-Fa-f]+$";
+        return color.matches(pattern) && (color.length() == 7 || color.length() == 9);
```

Comment:
@iamareebjamal The official documentation [here](https://developer.android.com/reference/android/graphics/Color.html#parseColor%28java.lang.String%29) states that ```Color.parseColor``` expects colors in ```#RRGGBB```
```#AARRGGBB``` format.
Hence i am using regular expressions to check if the color is one of those. If it isn't then setting it to null, so that default value will be used. Is it okay ?

Reply:
Rather than checking, do exception handling. Set header to light grey and text to black if there's any error

Reply:
Rather than coloring whole card, only color the header

Reply:
I am facing an issue here
This is my bind method

``` Java
public void bind(Track track) {
        try {
            track.color = "#nmnmnnm";
            binding.setTrack(track);
            binding.executePendingBindings();
        } catch (IllegalArgumentException illegalArgumentException) {
            track.color = null;
            track.fontColor = null;
        }
    }
```

I am trying to test the error and hence setting the ```track.color``` variable to an illegal argument. The exception is catched and then i set the color and fontColor to null which would mean that the layout file will use the default values which i have mentioned in it.
The problem is i am getting an empty list. 

Screenshot: 
 ![screenshot_2018-04-07-19-08-34-65](https://user-images.githubusercontent.com/21277837/38455528-2a4a9ecc-3a97-11e8-9ca1-1946cb4145aa.png)

I can't figure out the source of this problem.

Reply:
Because execute pending binding is not called in case of an exception

Reply:
Problem persists even after changing the ```bind``` method to the following
``` Java
public void bind(Track track) {
        try {
            track.color = "#nmnmnnm";
            binding.setTrack(track);
            binding.executePendingBindings();
        } catch (IllegalArgumentException illegalArgumentException) {
            track.color = null;
            track.fontColor = null;
            binding.setTrack(track);
            binding.executePendingBindings();
        }
    }
```

or 

``` Java
public void bind(Track track) {
        try {
            track.color = "#nmnmnnm";
            binding.setTrack(track);
            binding.executePendingBindings();
        } catch (IllegalArgumentException illegalArgumentException) {
            track.color = null;
            track.fontColor = null;
            binding.executePendingBindings();
        }
    }
```

Reply:
BTW, you shouldn't do this at all. Create two methods in presenter which take a track as an argument and return a color. This problem will be solved